records_mover/records/pandas/prep_for_csv.py

```diff
@@ -35,10 +34,13 @@ def _convert_series_or_index(series_or_index: T,
                                  'dateformat',
                                  records_format.hints)
                 pandas_date_format = '%Y-%m-%d'
-            if isinstance(series_or_index, pd.Series):
-                return series_or_index.dt.strftime(pandas_date_format)
+            if isinstance(series_or_index[0], pd.Timestamp):
+                if isinstance(series_or_index, pd.Series):
+                    return series_or_index.dt.strftime(pandas_date_format)
+                else:
+                    return series_or_index.strftime(pandas_date_format)
             else:
-                return series_or_index.strftime(pandas_date_format)
+                return series_or_index.apply(pd.Timestamp).dt.strftime(pandas_date_format)
```

Comment:
This bug was reproduced by the integration tests:

https://app.circleci.com/pipelines/github/bluelabsio/records-mover/1823/workflows/ef8242cd-55c3-4442-b72b-37bea71c89eb/jobs/15583


Which I wrote a component test for:

https://app.circleci.com/pipelines/github/bluelabsio/records-mover/1825/workflows/9249f424-617c-409e-a3de-e57d62ea811e/jobs/15587

That then revealed we had workaround expectation files that can be fixed to be more correct!

https://app.circleci.com/pipelines/github/bluelabsio/records-mover/1828/workflows/54774c27-e141-40ea-bb32-9d7123ff766d/jobs/15594